x/mongo/driver/crypt.go

```diff
@@ -293,7 +296,9 @@ func (c *crypt) decryptKey(ctx context.Context, kmsCtx *mongocrypt.KmsContext) e
 		addr = fmt.Sprintf("%s:%d", host, defaultKmsPort)
 	}
 
-	conn, err := tls.Dial("tcp", addr, &tls.Config{})
+	kmsProvider := kmsCtx.KMSProvider()
```

Comment:
Currently causes Evergreen failures due to Go not recognizing `KMSProvider()` as a method defined for `mongocrypt.KmsContext`. Is this to do with the libmongocrypt dependency update mentioned in [GODRIVER-2210](https://jira.mongodb.org/browse/GODRIVER-2210)

Reply:
So I don't think we'll have to "bump" the `libmongocrypt` dependency. We'll just have to update our documentation to say that version 1.3.0 of `libmongocrypt` is needed with version 1.8.0 and higher of the driver in GODRIVER-2210.

Reply:
That error indicates a version of libmongocrypt older than 1.3.0 is being included.

The [fetch-source](https://github.com/mongodb/mongo-go-driver/blob/master/.evergreen/config.yml#L75:L87) Evergreen function builds libmongocrypt the development branch on Linux / macOS and downloads the latest release tarball on Windows. Both cases should have 1.3.0 or newer.

If that error is still occurring, my initial suspicion is that leftover installations of libmongocrypt are interfering. I can help investigate if you have a link to a failing patch build handy.

Reply:
> If that error is still occurring, my initial suspicion is that leftover installations of libmongocrypt are interfering. I can help investigate if you have a link to a failing patch build handy.

My suspicion was incorrect. This was resolved by the addition of `KMSProvider` to `mongocrypt_kms_context_not_enabled.go` in 966f7e77767d2e45eb14fadfe62077397f9dc256